sklearn/linear_model/sag.py

```diff
@@ -256,13 +257,13 @@ def sag_solver(X, y, sample_weight=None, loss='log', alpha=1., beta=0.,
 
     # initialization
     if sample_weight is None:
-        sample_weight = np.ones(n_samples, dtype=np.float64, order='C')
+        sample_weight = np.ones(n_samples, dtype=X.dtype, order='C')
 
     if 'coef' in warm_start_mem.keys():
         coef_init = warm_start_mem['coef']
     else:
         # assume fit_intercept is False
-        coef_init = np.zeros((n_features, n_classes), dtype=np.float64,
+        coef_init = np.zeros((n_features, n_classes), dtype=X.dtype,
```

Comment:
This does not seem to be covered by any of the tests.

Reply:
@TomDLT as mentioned by @lesteve, this line is not covered since LogisiticRegression and Ridge always pass an array with initialized coefficients.

Is there a danger to remove this line? Is `sag_solver` could be used as a standalone function?